### PR TITLE
Add support for powerpc64

### PR DESCRIPTION
**Pull request by pkubaj**
This PR adds support for 64-bit POWER on big-endian.
There is already support for one 64-bit big-endian architecture (s390x).
There is also support for ppc64le (64-bit POWER little-endian).
So this PR shouldn't add much overhead.

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [ ] Does this PR change the the list of algorithms available -- either adding, removing, or renaming?  (If so, PRs in OQS-OpenSSL, OQS-BoringSSL, and OQS-OpenSSH will also be required by the time this is merged.)

### CODE REVIEW

**Comment by dstebila** [comment]
Thanks for the PR. Do you know of any resources for running continuous integration on this platform?

**Comment by pkubaj** [comment]
I don't think there are any that provide out-of-the-box. But e.g. OpenSSL crosscompiles for plenty of platforms: https://github.com/openssl/openssl/blob/master/.github/workflows/cross-compiles.yml

That said, I think it may be ok if liboqs builds on both ppc64le and s390x.

**Review by dstebila — APPROVED** [review]

**Comment by baentsch** [comment]
> I don't think there are any that provide out-of-the-box. But e.g. OpenSSL crosscompiles for plenty of platforms: https://github.com/openssl/openssl/blob/master/.github/workflows/cross-compiles.yml
> That said, I think it may be ok if liboqs builds on both ppc64le and s390x.

Indeed, the OpenSSL matrix also only tests s390x and ppc64le -- as does `liboqs` via Travis. If I'm not mistaken, the only BE PPC are IBM i/AIX machines... So, maybe [this link promising i/AIX "in the cloud"](https://help.skytap.com/skytap-on-ibm-cloud-overview.html) might be an option? Tagging @bhess for an opinion as to whether there's a way (& interest) to add IBM i/AIX to the `liboqs` CI.

**Comment by pkubaj** [comment]
Actually, all the POWER CPUs up to Power10 support BE. It's just that the current trend is to move to LE, but it's not like you can't run BE. In fact, I have a dual boot on my POWER9 workstation with both LE and BE.

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,6 +59,12 @@ elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc64le|powerpc64le")
     if(${OQS_DIST_BUILD})
         set(OQS_DIST_PPC64LE_BUILD ON)
     endif()
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "(ppc64|powerpc64)")
+    set(ARCH "ppc64")
```
Comment (baentsch) [inline CMakeLists.txt:63]: What about adding a warning message here about "suboptimal" CI for this platform?

**Comment by baentsch** [comment]
> Actually, all the POWER CPUs up to Power10 support BE. It's just that the current trend is to move to LE, but it's not like you can't run BE. In fact, I have a dual boot on my POWER9 workstation with both LE and BE.

Good to know. But if we can't do CI for the "generic" ppc64be platform I'd urge caution to adding it as an apparently fully supported platform to `liboqs`. See for example [this discussion](https://github.com/open-mpi/ompi/issues/4349) for what might ensue (say if some "big endian users" expected `liboqs` to work out of the box). 

So what about the suggestion to add a `cmake` warning message stating "this configuration not tested via continuous integration." (see comment above)?

**Review by baentsch — CHANGES_REQUESTED** [review]
Please add warning message about lack of CI for generic/big-endian PPC in CMakeLists.txt.

**Comment by bhess** [comment]
> Indeed, the OpenSSL matrix also only tests s390x and ppc64le -- as does `liboqs` via Travis. If I'm not mistaken, the only BE PPC are IBM i/AIX machines... So, maybe [this link promising i/AIX "in the cloud"](https://help.skytap.com/skytap-on-ibm-cloud-overview.html) might be an option? Tagging @bhess for an opinion as to whether there's a way (& interest) to add IBM i/AIX to the `liboqs` CI.

I believe most Linux distributions moved to a ppc64le target since little endian (switchable) is supported in POWER8 and later. Using qemu might be an option on Linux. AIX is a good point, but I don't think our CIs support an integration.

> So what about the suggestion to add a `cmake` warning message stating "this configuration not tested via continuous integration." (see comment above)?

Adding a warning is a good idea.

**Comment by pkubaj** [comment]
> I believe most Linux distributions moved to a ppc64le target since little endian (switchable) is supported in POWER8 and later. Using qemu might be an option on Linux. AIX is a good point, but I don't think our CIs support an integration.
Currently, Void Linux, Adelie and Gentoo support ppc64. Void and Adelie are ELFv2, but Gentoo is still ELFv1 on ppc64.
There are also FreeBSD and OpenBSD (both ELFv2).
This PR was done on FreeBSD.

File: CMakeLists.txt (hunk 1 above)
Comment (pkubaj) [reply CMakeLists.txt:64]: Done.

**Review by pkubaj — COMMENTED** [review]
Comment has been addressed.

**Review by baentsch — APPROVED** [review]

**Pull request merged**